### ISSUE TITLE
Adding tags via API ignores exclusive value for predicates

### ISSUE THREAD

**Issue by cudeso** [MISP/MISP#7318]
### Work environment

| Questions                 | Answers
|---------------------------|--------------------
| Type of issue             | Bug
| OS version (server)       | RedHat
| MISP version / git hash   | 2.4.141

### Expected behavior

When an event has a tag of a predicate of a taxonomy where the exclusive flag is set, adding an additional tag of that same predicate should fail. For example if workflow:state="incomplete" is set, adding workflow:state="complete" should fail.

### Actual behavior

Adding a tag via the API for a value of a predicate in a taxonomy ignores the "exclusive" setting.

### Steps to reproduce the behavior

```
>>> print(misp.get_event(uuid, pythonify=True).tags)
[<MISPTag(name=workflow:state="incomplete")>]
>>> print(misp.tag(uuid,'workflow:state="complete"'))
{'saved': True, 'success': True, 'name': 'Successfully attached 1 tags to Event (12230)', 'message': 'Successfully attached 1 tags to Event (12230)', 'url': '/tags/attachTagToObject'}
>>> print(misp.get_event(uuid, pythonify=True).tags)
[<MISPTag(name=workflow:state="incomplete")>, <MISPTag(name=workflow:state="complete")>]
```

**Comment by Wachizungu** [comment]
Yeah I spotted this earlier as well, note that this check is there for the other endpoints /attributes/addTag and /events/addTag. Just look for exclusiveTestPassed in the respective controllers.

There is a slight difference in implementation between those controllers as well (the way the response message is formulated is better in the attributes one I suppose).

Sidenote: Someone mentioned on gitter they were using this endpoint, adding a tag and then removing the other one after (tlp example), considering that... Changing this behavior might break some people's scripts. Anyway, I still think it's better to do so.

**Comment by cudeso** [comment]
> Sidenote: Someone mentioned on gitter they were using this endpoint, adding a tag and then removing the other one after (tlp example), considering that... Changing this behavior might break some people's scripts. Anyway, I still think it's better to do so.

It can indeed break some scripts, but imo the API should always behave the same as the UI.

**Comment by mokaddem** [comment]
This is indeed the intended behavior.
We decided to be more lax on the API than in the UI for the particular reason @Wachizungu mentioned. When we introduced the feature, we did not wanted to break existing workflows and we wanted users to bypass this restriction via the API.

**Comment by cudeso** [comment]
I understand it can break things. 
But this then means there are multiple sets of logic to follow, which isn't desirable.
 
It means that I can change add a tag (fe. 'complete' and 'incomplete') and publish the event via the API. Afterwards, when I add an attribute via the UI (and do no change whatsoever on the 'contextualisation') and want to publish this event, this is no longer possible. This will confuse users as the event comes from a 'published' state. This feels counter-inuitive.

**Comment by cudeso** [comment]
> But this then means there are multiple sets of logic to follow, which isn't desirable.

Maybe introducing a config option where you can decide whether or not to enforce this in the API could be a solution?
